pkg/agent/openflow/network_policy.go

```diff
@@ -150,7 +150,23 @@ type conjunctiveMatch struct {
 }
 
 func (m *conjunctiveMatch) generateGlobalMapKey() string {
-	return fmt.Sprintf("table:%d,type:%d,value:%s", m.tableID, m.matchKey, m.matchValue)
+	matchType := m.matchKey
+	valueStr := fmt.Sprintf("%s", m.matchValue)
+	// Use the unique format "x.x.x.x/xx" for IP address and IP net, to avoid generating two different global map keys
+	// for IP and IP/32. Use MatchDstIPNet/MatchSrcIPNet as match type to generate global cache key for both IP and IPNet.
+	if ip, ok := m.matchValue.(net.IP); ok {
+		valueStr = fmt.Sprintf("%s/32", ip.String())
+		switch m.matchKey {
+		case MatchDstIP:
+			matchType = MatchDstIPNet
+		case MatchSrcIP:
+			matchType = MatchSrcIPNet
+		}
+	}
+	if ipNet, ok := m.matchValue.(net.IPNet); ok {
+		valueStr = ipNet.String()
+	}
```

Comment:
L154 is for ofport case, L157 is for ip case, L167 is for ip net case, the code path should be exclusive for readability, for example:
```
	var valueStr string
	switch value := m.matchValue.(type) {
	case net.IP:
		valueStr = fmt.Sprintf("%s/32", value.String())
		switch m.matchKey {
		case MatchDstIP:
			matchType = MatchDstIPNet
		case MatchSrcIP:
			matchType = MatchSrcIPNet
		}
	case net.IPNet:
		valueStr = value.String()
	default:
		valueStr = fmt.Sprintf("%s", value)
	}
```
And better to explain why we need to convert IP to IP/32 in comments.

Reply:
I want to trigger more tests to see if the issue is really fixed, and after that I would update the link of the issue in this PR.